### PR TITLE
US77855 - When you select a course image in the ui, it should reflect that in the course tiles

### PR DESCRIPTION
**Pull request by jstefaniuk-d2l**
also revert the image in the case of a failure

Needs: https://git.dev.d2l/projects/CORE/repos/lp/pull-requests/3488/overview to work right

### CODE REVIEW

File: d2l-course-tile.html [hunk 1]
```diff
@@ -297,6 +297,25 @@
 				this.toggleClass('touch-menu-open', isOpen, this);
 				this.toggleClass('hover', isOpen, this);
 			},
+			setCourseImage: function(details) {
+				details = details || {};
+				var courseImage = this.$$('.course-image img');
+				var newImage = details.image.getLinkByClass('tile');
```
Comment (jstefaniuk-d2l) [inline d2l-course-tile.html]: Note: Use Trevor's stuff here

File: d2l-all-courses.html [hunk 2]
```diff
@@ -520,6 +520,9 @@ <h2 class="d2l-heading-3">{{localize('unpinnedCourses')}}</h2>
 			loadCourseTiles: function() {
 				this.delayCourseTileLoad = false;
 			},
+			setCourseImage: function(details) {
+				this.$$('d2l-course-tile-grid').setCourseImage(details);
```
Comment (apalaniuk) [inline d2l-all-courses.html]: Won't this always get the pinned courses grid?

File: d2l-course-tile.html [hunk 3]
```diff
@@ -406,6 +419,22 @@
...
 			},
+			_setCourseImageSrc: function() {
+				var href = '';
+
+				if (this._imageOverrideHref) {
+					href = this._imageOverrideHref;
+				} else {
+					var courseImageEntity = this._organization.getSubEntityByClass('course-image');
+					if (courseImageEntity) {
+						var courseImageLink = courseImageEntity.getLinkByRel('alternate');
```
Comment (jstefaniuk-d2l) [inline d2l-course-tile.html]: Note: Add Trevor's code here

File: d2l-all-courses.html (hunk 2 above)
Comment (jstefaniuk-d2l) [inline d2l-all-courses.html]: Here it always wants the pinned courses grid

**Comment by jstefaniuk-d2l** [comment]
Made a commit to fix unpinned images not getting updated in the all-courses view, I think it should work but can't test it out atm due to crappy network stuff at home

**Comment by cgalvind2l** [comment]
Everything seems to be working so far.

File: d2l-course-tile.html (hunk 1 above)
Comment (apalaniuk) [inline d2l-course-tile.html]: These comments mean, make sure to update `srcset` as well?

**Comment by apalaniuk** [comment]
Code LGTM

**Comment by jstefaniuk-d2l** [comment]
I'm going to be proactive and rebase Trevor's changes into this so that I don't have to deal with the rebase later today

**Comment by trevorjonesd2l** [comment]
LGTM

**Pull request merged**